tests/e2e_test/module_deletion_test.go

```diff
@@ -0,0 +1,157 @@
+package e2e_test
+
+import (
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/pkg/testutils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+var _ = Describe("Non Blocking Module Deletion", Ordered, func() {
```

Comment:
I suggest we can already apply some BDD test patterns with this, I saw on CLI [PR](https://github.com/kyma-project/cli/pull/1792/files#diff-dae65c740f15475708a1e4a92f0d63e65dbd56ea50108646cf4194be51758a76R11), we introduced decent example already, maybe @ameteiko can help in this case also.